File: PalindromeNumber.py
class Solution:
    # @return a boolean
    def isPalindrome(self, x):
        list = []
        #x = abs(x)
        if x<0:
            is_Palindrome = False
        else:
            is_Palindrome = True
        while x/10 >= 1:
            list.append(x%10)
            x = x/10
        list.append(x%10)

        for digit in range(0, len(list)/2+1):
            if list[digit] == list[-1-digit]:
                continue
            else:
                is_Palindrome = False
                break
        return is_Palindrome

sol = Solution()
#test = 1234321
#test = 2112
test = -2147483648
print sol.isPalindrome(test)

File: PalindromeNumber.cpp
// Return true if palindrome number
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    bool isPalindrome(int x) {
        vector<int> list;
        bool is_Palindrome;
        if(x<0) 
        {
            is_Palindrome = false;
        }
        else 
        {
            is_Palindrome = true;
        }
        while(x/10 >= 1)
        {
            list.push_back(x%10);
            x = x/10;
        }
        list.push_back(x%10);
        for (int i=0; i<list.size()/2+1; i++)
        {
            //cout << "digit "<<i<<" is "<< list[i] << ", digit "<<list.size()-i-1<<" is "<< list[list.size()-i-1] << endl;
            if(list[i] == list[list.size()-i-1])
            {
                continue;
            }
            else
            {
                is_Palindrome = false;
                break;
            }
        }
    return is_Palindrome;
    }
};

int main() {
    Solution sol;
    int input;
    cout << "Please enter a number" <<endl;
    cin>>input;
    cout<<"1 for palindrome number" <<endl;
    std::cout << sol.isPalindrome(input) << std::endl;
    return 0;
}

